presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1350,6 +1358,31 @@ else if (criteria instanceof JoinOn) {
             return output;
         }
 
+        private boolean needsJoinWarning(LogicalBinaryExpression expression)
+        {
+            ArrayDeque<Expression> queue = new ArrayDeque<Expression>();
```

Comment:
why do we need Deque here, a normal Queue is sufficient.

Reply:
Good point, I forgot `Dequeue` implements `Queue`, and we just need queue semantics.

Reply:
Fixed